Pills componet

Hasmik8 opened the pull request:
**Description**
I've created Pills component, which has two types: Basic (default,selected) and Light (success, danger, warning).
 In case of Basic, is has only one props, which is its children ,and in case of Light, it has both children and variant (success, danger, warning).

Basic Pills selected : 
![Снимок экрана от 2020-10-15 18-01-32](https://user-images.githubusercontent.com/46753920/96140112-a2ca6a80-0f10-11eb-98a0-ed35fd6246b0.png)
![Снимок экрана от 2020-10-15 18-03-27](https://user-images.githubusercontent.com/46753920/96140239-c68db080-0f10-11eb-88c4-f9cea485e71c.png)
![Снимок экрана от 2020-10-15 18-03-55](https://user-images.githubusercontent.com/46753920/96140282-cf7e8200-0f10-11eb-8e7c-84549fa91f80.png)

Light Pills : 

![Снимок экрана от 2020-10-15 18-04-40](https://user-images.githubusercontent.com/46753920/96140373-eb822380-0f10-11eb-9727-f00216045d05.png)
![Снимок экрана от 2020-10-15 18-04-57](https://user-images.githubusercontent.com/46753920/96140407-f50b8b80-0f10-11eb-9994-c88ce4dd6e7c.png)
![Снимок экрана от 2020-10-15 18-05-11](https://user-images.githubusercontent.com/46753920/96140445-ff2d8a00-0f10-11eb-8f1e-371375231b54.png)

Hasmik8 committed:
create pills component

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

Hasmik8 committed:
create pills component

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

S-ayanide commented on src/core/Pills/BasicPills/BasicPills.stories.tsx:
```diff
@@ -0,0 +1,26 @@
+  // Litmus Portal
+  .add('Litmus Portal', () => (
+    <ThemedBackground platform="litmus-portal" row>
+      <BasicPills>11 experiments</BasicPills>
```
Can we keep the name as `Basic Pill` here, similarly for all the platform demo Pill here in storybook, '11 experiments' is basically a dummy text.

S-ayanide commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,25 @@
+const BasicPills: React.FC<BasicPillsProps> = ({ children }) => {
+  const classes = useStyles();
+  const [selected, setSelected] = useState<selectType>(false);
+  const selectPills = (event: MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setSelected(!selected);
+  };
+  return (
+    <Button
+      onClick={selectPills}
```
Can we reduce the size of this and directly do 

```suggestion 
  return (
    <Button
      onClick={setSelected(!selected)}
```

Hasmik8 replied:
ok

S-ayanide commented on src/core/Pills/BasicPills/__tests__/BasicPills.test.tsx:
```diff
@@ -0,0 +1,16 @@
+      </KuberaThemeProvider>
+    );
+
+    expect(getByText('11 experiments')).toBeTruthy();
```
```suggestion
    expect(getByText('Basic Pill')).toBeTruthy();
```
Based on previous changes

S-ayanide commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,25 @@
+  return (
+    <Button
+      onClick={selectPills}
+      className={`${classes.basicPills} ${selected ? classes.selected : ''}`}
```
Can we do this instead! To follow the convention

```suggestion
      className={`${classes.root} ${selected ? classes.selected : ''}`}
```

S-ayanide commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,18 @@
+import { makeStyles, Theme } from '@material-ui/core';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  basicPills: {
```
```suggestion
  root: {
```

S-ayanide commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,18 @@
+
+const useStyles = makeStyles((theme: Theme) => ({
+  basicPills: {
+    height: '1.5rem',
```
We shouldn't hardcode the height, rather we can give a `padding`

S-ayanide commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,18 @@
+    height: '1.5rem',
+    borderRadius: '0.1875rem',
+    fontSize: '0.625rem',
+    lineHeight: '170%',
```
Line Height won't be required if we have a padding

S-ayanide commented on src/core/Pills/LightPills/LightPills.stories.tsx:
```diff
@@ -0,0 +1,32 @@
+  // Litmus Portal
+  .add('Litmus Portal', () => (
+    <ThemedBackground platform="litmus-portal" row>
+      <LightPills variant="success">Completed</LightPills>
+      <LightPills variant="warning">Runing</LightPills>
+      <LightPills variant="danger">Failed</LightPills>
```
```suggestion
      <LightPills variant="success">Success</LightPills>
      <LightPills variant="warning">Warning</LightPills>
      <LightPills variant="danger">Danger</LightPills>
```

Can you please make this change for al the storybook LightPills!

S-ayanide commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -0,0 +1,32 @@
+type Variant = 'success' | 'warning' | 'danger' | undefined;
+
+interface BasicPillsProps extends PillsBaseProps {
+  children: string;
```
`children` would be auto inferred thus no need to add it to interface

S-ayanide commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -0,0 +1,32 @@
+    }
+  };
+  return (
+    <Button className={`${classes.lightPills} ${getVariant(variant)}`}>
```
```suggestion
    <Button className={`${classes.root} ${getVariant(variant)}`}>
```

S-ayanide commented on src/core/Pills/LightPills/style.ts:
```diff
@@ -0,0 +1,25 @@
+import { makeStyles, Theme } from '@material-ui/core';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  lightPills: {
```
```suggestion
  root: {
```

S-ayanide commented on src/core/Pills/LightPills/style.ts:
```diff
@@ -0,0 +1,25 @@
+
+const useStyles = makeStyles((theme: Theme) => ({
+  lightPills: {
+    height: '1.5rem',
```
Can be replaced with `padding`

S-ayanide commented on src/core/Pills/LightPills/style.ts:
```diff
@@ -0,0 +1,25 @@
+    height: '1.5rem',
+    borderRadius: '0.1875rem',
+    fontSize: '0.625rem',
+    lineHeight: '130.2%',
```
Wouldn't be required if `padding` is present

Hasmik8 committed:
fix issues

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

Hasmik8 committed:
fix issues N1

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

S-ayanide commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,18 @@
+  const classes = useStyles();
+  const [selected, setSelected] = useState<selectType>(false);
+  return (
+    <Button
+      onClick={() => setSelected(!selected)}
+      className={`${classes.root} ${selected ? classes.selected : ''}`}
+    >
+      {children}
+    </Button>
```
I think the pills shouldn't have an onClick toggler. It has two states, `default` - when not selected and `active` - when selected

When the pill is normal `root` class will have base styling
When the pills are selected `selected` property of the button would be overridden. (Just like hover)

Basically, pills would never be clicked/clickable, they can either take a prop as `selected` or not, or you could follow an overridden approach based on a div. Wherever the pills are attached (a card, template, etc) if that particular div/box is selected the pills should be active (not by directly clicking on the pill).

S-ayanide commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -0,0 +1,31 @@
+    }
+  };
+  return (
+    <Button className={`${classes.root} ${getVariant(variant)}`}>
+      {children}
+    </Button>
```
This won't be clickable either. And should not be hoverable/selected either

Hasmik8 committed:
fix issues N2

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

ajeshbaby commented:
@arkajyotiMukherjee @S-ayanide Should the components have for Kubera-Core and Kubera-Portal as well?

arkajyotiMukherjee commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,18 @@
+    background: theme.palette.primary.light,
+    color: theme.palette.text.secondary,
+    '&:hover': {
+      background: theme.palette.primary.main,
```
pills are just for information...hover and selected states won't be required for this component

arkajyotiMukherjee commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { useStyles } from './style';
+
+const BasicPills: React.FC = ({ children }) => {
```
can you change the name to just `Pills`

arkajyotiMukherjee commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -0,0 +1,28 @@
+type Variant = 'success' | 'warning' | 'danger' | undefined;
+
+interface LightPillsProps extends PillsBaseProps {
+  variant?: Variant;
```
```suggestion
  variant: Variant;
```
This variant will be a compulsory parameter

arkajyotiMukherjee commented on src/core/Pills/base.ts:
```diff
@@ -0,0 +1,5 @@
+import { ButtonProps } from '@material-ui/core/Button';
+
+type PillsBaseProps = Omit<ButtonProps, 'variant'>;
```
Pills don't need to extend from Button...try this instead
https://material-ui.com/components/chips/#chip
fix the `variant` from Mui to outlined and we'll have out own one...so Omit<ChipProps, 'variant'> will be the used

S-ayanide commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,18 @@
+    textTransform: 'none',
+    background: theme.palette.primary.light,
+    color: theme.palette.text.secondary,
+    '&:hover': {
+      background: theme.palette.primary.main,
+    },
```
Hover state is not there in Plls.

selected: {} would be there which would be inferred from the card/div background selection

S-ayanide replied:
I think you'd require a variant {default, selected}. Based on the props passed by the environment, you'd need to change the css property.

arkajyotiMukherjee commented:
> @arkajyotiMukherjee @S-ayanide Should the components have for Kubera-Core and Kubera-Portal as well?

Yes they'll be added...with upcoming PRs

Hasmik8 committed:
fix issues N3

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

Hasmik8 committed:
fix issues N3

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

S-ayanide commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,24 @@
+
+const Pills: React.FC<PillsProps> = ({ variant, children }) => {
+  const classes = useStyles();
+  const getVariant = (variant: Variant) => {
+    switch (variant) {
+      case 'selected':
+        return classes.active;
+      default:
+        return classes.default;
+    }
+  };
```
```suggestion
  const getVariant = (variant: Variant) => {
    switch (variant) {
      case 'selected':
        return classes.active;
      default:
        return '';
    }
  };
```

S-ayanide commented on src/core/Pills/BasicPills/style.ts:
```diff
@@ -0,0 +1,20 @@
+import { makeStyles, Theme } from '@material-ui/core';
+
+const useStyles = makeStyles((theme: Theme) => ({
+  root: {
+    padding: theme.spacing(0.45, 1.5),
+    borderRadius: '0.1875rem',
+    fontSize: '0.625rem',
+    fontWeight: 500,
+    textTransform: 'none',
+    color: theme.palette.text.secondary,
+  },
+  default: {
+    background: theme.palette.primary.light,
+  },
```
```suggestion
  root: {
    padding: theme.spacing(0.45, 1.5),
    borderRadius: '0.1875rem',
    fontSize: '0.625rem',
    fontWeight: 500,
    textTransform: 'none',
    color: theme.palette.text.secondary,
    background: theme.palette.primary.light,
  },
```

Hasmik8 committed:
fix

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

S-ayanide reviewed (dismissed):
LGTM

arkajyotiMukherjee commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,24 @@
+
+type Variant = 'default' | 'selected';
+
+interface PillsProps {
```
```suggestion
interface PillsProps extends PillsBaseProps {
```

arkajyotiMukherjee commented on src/core/Pills/BasicPills/BasicPills.tsx:
```diff
@@ -0,0 +1,24 @@
+    }
+  };
+  return (
+    <div className={`${classes.root} ${getVariant(variant)}`}>{children}</div>
```
now we need to change the div to a Chip...as we extended the base props from chip

arkajyotiMukherjee replied:
https://material-ui.com/components/chips/#chip
take a look to see the options available and change acc to need.
Omit variant and colors from the interface as we can give our own variant

Hasmik8 commented:
@arkajyotiMukherjee I replaced the div with chip Material UI component, there an issue appeared, I'm not able to take the child from the Pills component. That's why I added props to Pills component, the name is label. Or I should add the child in to the interfaceand set the type string. 
![Снимок экрана от 2020-10-16 17-47-26](https://user-images.githubusercontent.com/46753920/96266435-b2f94d00-0fd7-11eb-9aff-e69f55db526c.png)
![Снимок экрана от 2020-10-16 17-47-33](https://user-images.githubusercontent.com/46753920/96266446-b5f43d80-0fd7-11eb-932d-8ae11fa0e5d6.png)

arkajyotiMukherjee commented:
@Hasmik8 yes you can definitely use label as it has a different API...no need to use children here

Hasmik8 committed:
fix pills component

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

arkajyotiMukherjee commented on src/core/Pills/base.ts:
```diff
@@ -1,5 +1,5 @@
 import { ChipProps } from '@material-ui/core/Chip';
 
-type PillsBaseProps = Omit<ChipProps, 'variant'>;
+type ChipBaseProps = Omit<ChipProps, 'variant' | 'label'>;
```
label is already there...no need to omit

arkajyotiMukherjee commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -1,13 +1,15 @@
-interface LightPillsProps extends PillsBaseProps {
+interface LightPillsProps extends ChipBaseProps {
   variant: Variant;
+  label: string;
```
label is already present in BaseProps...no need to add your own

arkajyotiMukherjee reviewed (changes requested):
Also can you update your branch

S-ayanide commented:
Can you please update your branch to the latest which has Kubera Portal and Kuber Core themes too. Also, update your stories accordingly.

Hasmik8 committed:
update branch

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

arkajyotiMukherjee commented on src/core/Pills/base.ts:
```diff
@@ -0,0 +1,3 @@
+import { ChipProps } from '@material-ui/core/Chip';
+
+export type ChipBaseProps = Omit<ChipProps, 'variant' | 'label'>;
```
```suggestion
export type ChipBaseProps = Omit<ChipProps, 'variant'>;
```

arkajyotiMukherjee commented on src/core/Pills/LightPills/LightPills.tsx:
```diff
@@ -0,0 +1,30 @@
+
+interface LightPillsProps extends ChipBaseProps {
+  variant: Variant;
+  label: string;
```
```suggestion

```

arkajyotiMukherjee reviewed (changes requested):
This review was not addressed @Hasmik8

Hasmik8 committed:
fix

Signed-off-by: Hasmik8 <hakobyan.hasmik05@gmail.com>

Hasmik8 committed:
Merge branch 'dev' of https://github.com/mayadata-io/kubera-ui into pills-componet

AVRahul reviewed (approved):
LGTM

arkajyotiMukherjee reviewed (approved):
lgtmt
